src_plugins/create/ppx_deriving_create.cppo.ml

```diff
@@ -7,30 +7,41 @@ open Ppx_deriving.Ast_convenience
 let deriver = "create"
 let raise_errorf = Ppx_deriving.raise_errorf
 
-let parse_options options =
-  options |> List.iter (fun (name, expr) ->
-    match name with
-    | _ -> raise_errorf ~loc:expr.pexp_loc "%s does not support option %s" deriver name)
+let attr_default context = Attribute.declare "deriving.create.default" context
+  Ast_pattern.(single_expr_payload __) (fun e -> e)
+let ct_attr_default = attr_default Attribute.Context.core_type
+let label_attr_default = attr_default Attribute.Context.label_declaration
 
-let attr_default attrs =
-  Ppx_deriving.(attrs |> attr ~deriver "default" |> Arg.(get_attr ~deriver expr))
+let attr_split context = Attribute.declare "deriving.create.split" context
+  Ast_pattern.(pstr nil) ()
+let ct_attr_split = attr_split Attribute.Context.core_type
+let label_attr_split = attr_split Attribute.Context.label_declaration
 
-let attr_split attrs =
-  Ppx_deriving.(attrs |> attr ~deriver "split" |> Arg.get_flag ~deriver)
+let attr_main context = Attribute.declare "deriving.create.main" context
+  Ast_pattern.(pstr nil) ()
+let ct_attr_main = attr_main Attribute.Context.core_type
+let label_attr_main = attr_main Attribute.Context.label_declaration
+
+let attribute_get2 attr1 x1 attr2 x2 =
+  match Attribute.get attr1 x1, Attribute.get attr2 x2 with
+  | Some _ as y, _ -> y
+  | None, y -> y
```

Comment:
Looking at this, I'm not sure what the value is of allowing the attribute to be attached to the `core_type` in addition to the `label_declaration`. It seems that they all should be attached to the label declaration since the deriver requires more than just the type to properly interpret them. Attaching the attribute to the core_type requires extra care from the user so it's unlikely that this is a common usecase.

I understand that we need to preserve this behaviour here but I'm not sure a lot of other ppxlib users would benefit from this.

Reply:
Most uses of this disappeared by using `has_flag`, but a few still remained where the actual value is being extracted.

I recall from somewhere that this is a rather common thing to allow attributes on record field types without having to add parenthesis, the attribute is on the row label instead then I think. I haven't checked but I think this might be similar in more modern derivers as well.